source/extensions/network/dns_resolver/apple/apple_dns_impl.h

```diff
@@ -125,5 +126,20 @@ class AppleDnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logg
   AppleDnsResolverStats stats_;
 };
 
+
+class AppleDnsResolverFactoryImpl : public DnsResolverFactory {
+public:
+  std::string name() const override { return "envoy.dns_resolver.apple"; }
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return ProtobufTypes::MessagePtr{new Envoy::ProtobufWkt::Struct()};
```

Comment:
See above comment on unique messages (even when empty) for each extension type.

Reply:
ACK